draft-wang-ppm-differential-privacy.md

```diff
@@ -403,31 +403,266 @@ This section describes various mechanisms required for implementing DP
+~~~
+class DpMechanism:
+    DataType
+    DebiasedDataType
+
+    def add_noise(self, data: DataType) -> DataType:
+        """Add noise to a piece of input data. """
+        pass
 
-The section defines a generic interface for DP policies VDAFs.
+    def sample_noise(self, num_reps: int, dimension: int) -> DataType:
+        """
+        Sample noise for `num_reps` number of times,
+        with `dimension`.
+        """
+        pass
 
-> TODO: Junye to spell out the interface of dp policies, "Noise Mechanisms to Accomplish DP"
+    def debias(self,
+               agg: DataType,
+               meas_count: int) -> DebiasedDataType:
+        """
+        Debias the data due to the added noise.
+        This doesn't apply to all noises. Some Client-DP mechanisms
+        need this functionality.
+        """
+        pass
+~~~
 
-## Client-DP for Prio3Histogram
+## Discrete Laplace
 
-> TODO: Concretely describe a Client-DP mechanism for Prio3Histogram.
+> TODO: Specify a Laplace sampler
 
-## Aggregator-DP for Prio3Histogram
+## Discrete Gaussian {#discrete-gaussian}
+
+> TODO: Specify a Gaussian sampler
+
+## Randomized Response {#rr}
+
+> TODO: Specify any mechanisms required for randomized response mechanisms
+
+# DP Policies for VDAFs {#policies}
 
-> TODO: Concretely describe an Aggregator-DP mechanism for Prio3Histogram.
+The section defines a generic interface for DP policies for VDAFs.
+
+We will define an interface `DpPolicy` that composes the following:
+
+* An optional Client-DP mechanism that adds noise to Clients' measurements.
+
+* An optional Aggregator-DP mechanism that adds noise to an Aggregator's
+  aggregate share, based on the number of measurements, and the minimum
+  batch size.
+
+* An optional debiasing step that removes the bias in DP mechanisms (i.e.
+  `DpMechanism.debias`).
+
+The composition of Client- and Aggregator-DP mechanisms defines the DP
+policy for a VDAF, and enforces the DP guarantee.
+
+~~~
+class DpPolicy:
+    Measurement
+    AggregateShare
+    AggregateResult
+    DebiasedAggregateResult
+
+    def add_noise_to_measurement(self,
+                                 meas: Measurement,
+                                 ) -> Measurement:
+        """
+        Add noise to measurement, if required by the Client-DP
+        mechanism.
+        """
+        pass
+
+    def add_noise_to_agg_share(self,
+                               agg_share: AggregateShare,
+                               meas_count: Unsigned,
+                               min_batch_size: Unsigned,
+                               ) -> AggregateShare:
+        """
+        Add noise to aggregate share, if required by the Aggregator-DP
+        mechanism.
+        """
+        pass
+
+    def debias_agg_result(self,
+                          agg_result: AggregateResult,
+                          meas_count: Unsigned,
+                          min_batch_size: Unsigned,
+                          num_aggregators: Unsigned,
```

Comment:
@cjpatton Added `num_aggregators` parameter in this method, which is needed by `Prio3SumVecWithRandomizedResponse`.